src/shared/components/InfiniteScrollWrapper/InfiniteScrollWrapper.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { useEffect, useState } from 'react';
+
+export default function withInfiniteScroll(WrappedComponent) {
+  return function ComponentWithScroll(props) {
+    // eslint-disable-next-line react/prop-types
+    const { doLoadData } = props;
+    const [page, setPage] = useState(1);
+
+    useEffect(() => {
+      doLoadData({ page }, false);
+    }, [page, doLoadData]);
+
+    let isBlock = false;
+
+    const onScroll = () => {
+      const windowHeight = window.innerHeight;
+      const documentHeight = document.body.clientHeight;
+      const scrollTop = window.scrollY;
+
+      if (windowHeight + scrollTop > documentHeight + 30 && !isBlock) {
```

Comment:
Here's a simple example of handling this logics
https://www.voidcanvas.com/implementing-infinite-scroll-with-react-js-without-plugin/